src/ContrastRestClient/Http/ContrastRestClient.cs

```diff
@@ -47,25 +48,33 @@ public Stream GetResponseStream(string apiEndpoint)
             var responseTask = _httpClient.GetAsync(apiEndpoint);
             responseTask.Wait();
 
-            CheckResponseStatusCode(apiEndpoint, responseTask.Result.StatusCode);
+            CheckResponse(apiEndpoint, responseTask.Result);
             
             var responseStreamTask = responseTask.Result.Content.ReadAsStreamAsync();
             responseStreamTask.Wait();
 
             return responseStreamTask.Result;
         }
 
-        private static void CheckResponseStatusCode(string apiEndpoint, System.Net.HttpStatusCode statusCode)
+        private static void CheckResponse(string apiEndpoint, System.Net.Http.HttpResponseMessage result)
         {
-            if ((int)statusCode >= 300)
+            if ((int)result.StatusCode >= 300)
             {
-                if (statusCode == System.Net.HttpStatusCode.NotFound)
+                if (result.StatusCode == System.Net.HttpStatusCode.NotFound)
                 {
-                    throw new ResourceNotFoundException("Resource: '" + apiEndpoint + "' not found.");
+                    throw new ResourceNotFoundException($"Resource: '{apiEndpoint}' not found.");
                 }
-                else
+                else if (result.StatusCode == System.Net.HttpStatusCode.Found 
+                    && result.Headers.Contains("Location")
+                    && result.Headers.GetValues("Location").First().EndsWith("/Contrast/unauthorized.html") )
```

Comment:
TeamServer can execute on both Windows and Linux servers, therefore, it might be best to use `StringComparison.OrdinalIgnoreCase` for the `EndsWith` check.